src/model/openthaigpt_pretraining_model/lightning/utils.py

```diff
@@ -106,17 +108,16 @@ def __init__(
             precision=precision,
         )
         self.fabric.launch()
+
         if model_name == "llama":
             model_name = LLAMA_MODEL  # for tokenizer
-            cfg = LlamaConfig(
+            self.model = model = make_model_llama(
                 vocab_size=vocab_size,
-                hidden_size=1024,
-                num_hidden_layers=8,
-                num_attention_heads=8,
-                hidden_act="silu",
-                max_position_embeddings=context_length,
+                context_length=context_length,
+                use_checkpointing=checkpoint,
+                checkpoint_only_attentions=checkpoint_only_attention,
```

Comment:
`checkpoint_only_attentions` or `checkpoint_only_attention` you should use one.

Reply:
done